tests/PySys/local_mqtt_pub_sub/run.py

```diff
@@ -18,26 +18,37 @@
 class PySysTest(BaseTest):
     def execute(self):
         tedge = "/usr/bin/tedge"
+        sudo = "/usr/bin/sudo"
 
         sub = self.startProcess(
-            command=tedge,
-            arguments=["mqtt", "sub", "atopic"],
+            command=sudo,
+            arguments=[tedge, "mqtt", "sub", "atopic"],
             stdouterr="tedge_sub",
             background=True,
         )
 
+        # Wait for a small amount of time to give tedge sub time
+        # to initialize
         time.sleep(0.1)
 
         pub = self.startProcess(
-            command=tedge,
-            arguments=["mqtt", "pub", "atopic", "amessage"],
-            stdouterr="tedge_pub",
+            command=sudo,
+            arguments=[tedge, "mqtt", "pub", "atopic", "amessage"],
+            stdouterr="tedge_pub2",
         )
 
         pub = self.startProcess(
-            command=tedge,
-            arguments=["mqtt", "pub", "atopic", "the message"],
-            stdouterr="tedge_pub",
+            command=sudo,
+            arguments=[tedge, "mqtt", "pub", "atopic", "the message"],
+            stdouterr="tedge_pub3",
+        )
+
+        # Kill the subscriber process explicitly with sudo as PySys does
+        # not have the rights to do it
+        kill = self.startProcess(
```

Comment:
One doesn't need to run `tedge mqtt` as sudo.

Reply:
I wasn't sure in this case. A strange thing that I observed there is an error when I run tedge pub with PySys:
"Error: failed to read the tedge configuration
Caused by:
User's Home Directory not found."
It works with sudo, probably PySys plays a bit around with the environment...